src/renderer.ts

```diff
@@ -153,6 +156,470 @@ export class DefaultCellRenderer implements ICellRendererFactory {
+
+    const cellDimension = col.calculateCellDimension(col.getWidth());
+    const colorScale = col.getColor();
+    const padding = context.option('rowPadding', 1);
+    return {
+
+      template: `<g class="heatmapcell"></g>`,
+      update: (n: SVGGElement, d: IDataRow, i: number) => {
+        const rect = d3.select(n).selectAll('rect').data(col.getValue(d.v, d.dataIndex));
+
+        rect.enter().append('rect');
+        rect
+          .attr({
+            y: padding,
+            x: (d, i) => (i * cellDimension),
+            width: cellDimension,
+            height: context.rowHeight(i),
+            fill: (d, i) => colorScale(d)
+          });
+        rect.exit().remove();
+      }
+    };
+  }
+
+
+  createHTML(col: MultiValueColumn, context: IDOMRenderContext): IHTMLCellRenderer {
+
+    const cellDimension = col.calculateCellDimension(col.getWidth());
+    const padding = context.option('rowPadding', 1);
+    const colorScale = col.getColor();
+
+    return {
+      template: `<div class="heatmapcell" style="top:${padding}px;">
+                                   </div>`,
+      update: (n: HTMLDivElement, d: IDataRow, i: number) => {
+        const g = d3.select(n);
+        const div = g.selectAll('div').data(col.getValue(d.v, d.dataIndex));
+        div.enter().append('div');
+        div
+          .style({
+            'width': cellDimension + 'px',
+            'background-color': (d, i) => colorScale(d),
+            'height': context.rowHeight(i) + 'px',
+            'left': (d, i) => (i * cellDimension) + 'px'
+          });
+
+
+      }
+    };
+  }
+
+  createCanvas(col: MultiValueColumn, context: ICanvasRenderContext): ICanvasCellRenderer {
+
+    const cellDimension = col.calculateCellDimension(col.getWidth());
+    const padding = context.option('rowPadding', 1);
+    const colorScale = col.getColor();
+    return (ctx: CanvasRenderingContext2D, d: IDataRow, i: number) => {
+
+      var data = col.getValue(d.v, d.dataIndex);
+      data.forEach(function (d, i) {
+        var x = (i * cellDimension);
+        ctx.beginPath();
+        ctx.fillStyle = colorScale(d);
+        ctx.fillRect(x, padding, cellDimension, context.rowHeight(i));
+      });
+    };
+  }
+
+}
+
+
+class SparklineCellRenderer implements ICellRendererFactory {
+
+
+  createSVG(col: MultiValueColumn, context: IDOMRenderContext): ISVGCellRenderer {
+
+    var xScale: any = col.getSparkLineXScale().range([0, col.getWidth()]);
+    var yScale: any = col.getSparkLineYScale();
+
+    return {
+
+      template: `<g class="sparklinecell"></g>`,
+      update: (n: SVGGElement, d: IDataRow, i: number) => {
+        const data = col.getValue(d.v, d.dataIndex);
+        yScale.range([context.rowHeight(i), 0]);
+        var line = d3.svg.line<number>()
+          .x((d, j) => xScale(j))
+          .y((d, j) => yScale(d))
+          .interpolate('linear');
+
+        let path = d3.select(n).select('path');
+        if (path.size() === 0) {
```

Comment:
why not adding the `path` element already to the template? which related to a similar question: why do I need a parent `g class="splarlinecell"` element at all?